substrate/frame/revive/src/lib.rs

```diff
@@ -891,26 +1060,38 @@ pub mod pallet {
 			gas_limit: Weight,
 			#[pallet::compact] storage_deposit_limit: BalanceOf<T>,
 			data: Vec<u8>,
+			signed_transaction: TransactionSigned,
```

Comment:
Transaction payload size is needed to benchmark on_finalize part per transaction, see [here](https://github.com/paritytech/polkadot-sdk/pull/9496/files#diff-f55a8416f5e8564cb4463b5f11dfd00d7c865aabab747a05887feeedd4e1c81dR1032-R1041).

So far in previous versions transaction payload was provided, and I was simply using this approach:
```
 .saturating_add(T::WeightInfo::on_finalize_block_per_tx(payload.len() as u32))
```
With this change we would have to encode the transaction to get the payload size.
```
.saturating_add(T::WeightInfo::on_finalize_block_per_tx(signed_transaction.signed_payload().len() as u32))
```

Not sure if this is ok taking into account that it will be encoded again  in `on_finalize`, when building Eth block.

Reply:
This is risky. Cause the pre-dispatch weight is then unknown to the block builder before including the tx. Especially since the payload is uncapped (we should probably change that).

If you need the rlp encoding in the weight annotation, why don't you pass that as a parameter? Then you don't need to encode there.